### PR TITLE
add LRU cache for transaction info

### LINKED ISSUE #70: `eth_getTransactionByHash` is not being cached in `getTransactionInfo`
There is a potential for `getTransactionInfo` to be called many times with the same transaction hash. Currently, this will result in a `eth_getTransactionByHash` RPC call every single time. A potential performance win could be caching this response instead.

https://github.com/hyperledger/firefly-evmconnect/blob/dd7b14d623a32660dd61d92f7ea5f5a9dd0e8a49/internal/ethereum/get_receipt.go#L76

### PR DESCRIPTION
**Pull request by shorsher**
- also removes unused TTL config option

closes #70

### CODE REVIEW

File: internal/ethereum/get_receipt.go [hunk 1]
```diff
@@ -75,11 +75,17 @@ type txInfoJSONRPC struct {
...
+	if ok {
+		return cached.(*txInfoJSONRPC), nil
+	}
+
 	rpcErr := c.backend.CallRPC(ctx, &txInfo, "eth_getTransactionByHash", hash)
 	var err error
 	if rpcErr != nil {
 		err = rpcErr.Error()
 	}
+	c.txCache.Add(hash.String(), txInfo)
```
Comment (awrichar) [inline internal/ethereum/get_receipt.go:88]: Should probably only cache if there was no error

File: internal/msgs/en_config_descriptions.go [hunk 2]
```diff
@@ -38,4 +38,6 @@ var (
 	ConfigEventsCatchupThreshold      = ffc("config.connector.events.catchupThreshold", "How many blocks behind the chain head an event stream or listener must be on startup, to enter catchup mode", i18n.IntType)
 	ConfigEventsCheckpointBlockGap    = ffc("config.connector.events.checkpointBlockGap", "The number of blocks at the head of the chain that should be considered unstable (could be dropped from the canonical chain after a re-org). Unless events with a full set of confirmations are detected, the restart checkpoint will this many blocks behind the chain head.", i18n.IntType)
 	ConfigEventsFilterPollingInterval = ffc("config.connector.events.filterPollingInterval", "The interval between polling calls to a filter, when checking for newly arrived events", i18n.TimeDurationType)
+	ConfigTxCacheSize                 = ffc("config.connector.txCacheSize", "Maximum of transactions to hold in the transaction info cache", i18n.IntType)
+	ConfigTxCacheTTL                  = ffc("config.connector.txCacheTTL", "Time to live for the transaction info cache", i18n.TimeDurationType)
```
Comment (awrichar) [inline internal/msgs/en_config_descriptions.go:42]: Not used?

Comment (awrichar) [reply internal/msgs/en_config_descriptions.go:42]: Can probably prune `ConfigBlockCacheTTL` too

**Review by awrichar — APPROVED** [review]

File: internal/msgs/en_config_descriptions.go (hunk 2 above)
Comment (shorsher) [reply internal/msgs/en_config_descriptions.go:42]: ah thought I removed, thank you

**Pull request merged**